tests/integration/targets/mso_schema_template_anp_epg/tasks/main.yml

```diff
@@ -814,10 +830,31 @@
   - name: Verify Add description
     assert:
       that:
-      - add_epg_desc is changed
-      - add_epg_desc.current.description == "Description of ANP EPG"
+        - add_epg_desc is changed
+        - add_epg_desc.current.description == "Description of ANP EPG"
+
+- name: Execute tasks only for MSO version >= 4.0
+  when:
+  - version.current.version is version('4.0', '>=')
+  block:
+  # TODO Determine which asserts are needed for this case if this case is still needed.
+  # Test cases for version greater than 3.3 and less than 4.0 involve only schema_validate which is deprecated
+  - name: Add EPG service type parameters (for version greater than 4.0)
+    # Add EPG with service_type: 'Azure-Storage' without triggering validation rule
+    mso_schema_template_anp_epg:
+      <<: *new_epg
+      epg_type: 'service'
+      deployment_type: 'third_party'
+      service_type: 'Azure-Storage'
+      access_type: 'private'
 
-  - name: Add EPG service type parameters (for version greater than 3.3)
+# Add EPG when MSO version >= 3.3 and < 4.0
+- name: Execute tasks only for MSO version >= 3.3 and < 4.0
+  when:
+  - version.current.version is version('3.3', '>=')
+  - version.current.version is version('4.0', '<')
+  block:
+  - name: Add EPG service type parameters (for version greater than 3.3 and less than 4.0)
```

Comment:
yes this is still needed because the schema validate even though it is back will never be reached in 4.0, since the validation is already triggered upon request thus not allowing the wrong service type parameters